pcsx2: 1.7.2105 -> 1.7.2731, support Vulkan

SuperSamus opened the pull request:
###### Description of changes

Bump version (there have been lots of additions and improvements), add Vulkan support.

###### Things done

Both OpenGL on Vulkan work on X11/XWayland.

Trying to launch a game on Wayland crashes the application even on the software renderer, with no error message explaining what happened (the most I could get is `Gdk-Message: 17:36:53.652: Error flushing display: Broken pipe`). This is a regression.

It's not something to lose sleep over since opening PCSX2 with the .desktop file sets `GDK_BACKEND=x11` and the project is also moving to Qt, but it's still the reason I'm opening this as a draft.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

SuperSamus committed:
pcsx2: 1.7.2105 -> 1.7.2731, support Vulkan

SuperSamus commented:
32 bit support is now removed.

Still crashes on Wayland.

ajs124 commented:
Does the release currently in nixpkgs work on wayland?

SuperSamus commented:
It does.

SuperSamus commented:
Tried the newest version, it still doesn't work with Wayland.

ajs124 commented:
Hm, ok. So we'll just leave this as a draft until it does?

You mentioned they are moving to Qt, so do you think it will be after that?

I honestly don't know anything about this project, I'm just here because I would/will merge this.

SuperSamus commented:
I forced the use of X11 (and added a comment about that).
Also, I added a few dependencies that would otherwise be bundled.
If you think that this regression isn't important (plus the loss of 32 bit support, that one is not supported anymore upstream), then this can be merged.

ajs124 commented:
Would be nice to get some input from the people listed as maintainers. If one of them uses this on wayland, I wouldn't want to break that.

hrdinka commented:
If it does work with `GDK_BACKEND=x11` on Wayland, I am in favor of merging the PR. The other changes are a great improvements. Since PCSX2 has always been a bit of a problem child, especially the move to x86_64 will help a lot.

In case it does not work on wayland at all—I am not running Wayland (nvidia) and cannot test/fix it—it depends on how soon the transition to QT will be made. If it is soon then let’s wait, if it is going to take longer than lets merge it and maybe someone else can help fixing it.

If I understood it correctly, it does work on Wayland by forcing the use of XWayland? If so, lets merge it.

SuperSamus commented:
Yes, `GDK_BACKEND=x11` forces XWayland, making it work on a Wayland environment.

hrdinka commented:
Great then I am in favor of merging it. Let‘s wait for the checks to finish. Thanks to everyone for the work done here :+1:
